src/main/java/gov/ornl/ssm/ml/ui/OIDCLoginSecurityConfig.java

```diff
@@ -0,0 +1,42 @@
+package gov.ornl.ssm.ml.ui;
+
+import java.net.URI;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.oauth2.client.oidc.web.logout.OidcClientInitiatedLogoutSuccessHandler;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.logout.LogoutSuccessHandler;
+
+/**
+ * Security configuration for OIDC sign in.
+ * 
+ * @author Robert Smith
+ *
+ */
+@Configuration
+public class OIDCLoginSecurityConfig {
+	
+	@Autowired
+	private ClientRegistrationRepository clientRegistrationRepository;
+	
+	@Bean
+	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+	    
+		// Return to homepage on logout
+		OidcClientInitiatedLogoutSuccessHandler handler = new OidcClientInitiatedLogoutSuccessHandler(clientRegistrationRepository);
+		handler.setPostLogoutRedirectUri("http://localhost:8080/machine-learning/");
+		
+		// Permit only oauth authorized requests
+		http.authorizeRequests(authorizeRequests -> authorizeRequests
+	    	.anyRequest().authenticated())
+	    	.oauth2Login(e -> e.permitAll())
+	    	.logout(e -> e.logoutSuccessHandler(handler));
+		http.csrf().disable();
+	    return http.build();
+	}
```

Comment:
Oh cool, so this is the OIDC with the new security filter chain?

I haven't completely switched the catalog over yet for the OIDC part so good to see this.

Question: do we have tests around the OIDC part yet?

Reply:
I've never had any tests on this. Generally, I've always found that GUI tests have never found a real issue in a program I've ever worked with, but instead keep breaking the build when the tests can't find something that gets an internal name/path slightly changed.

Reply:
Gotcha. thanks